app/controllers/public/sessions.js

```diff
@@ -13,6 +14,32 @@ export default class SessionsController extends Controller {
   timezone = null;
   preserveScrollPosition = true;
 
+  @computed('model.session.@each', 'timezone')
+  get groupByDateSessions() {
+    let sessions = groupBy(this.model.session.toArray(), s => moment.tz(s.startsAt, this.timezone).format('dddd, Do MMMM'));
+
+    if (this.sort === 'title') {
+      sessions = groupBy(this.model.session.toArray(), '');
+    }
+
+    const arr = [];
+
+    for (const key in sessions) {
+      let date = key;
+
+      if (date === 'undefined') {
+        date = null;
+      }
+
+      arr.push({
+        date,
+        'sessions': sessions[key]
+      });
+    }
+
+    return arr;
```

Comment:
```suggestion
    return Object.entries(sessions).map((date, sessions) => ({ date, sessions }));
```

And remove the loop from above

Reply:
This was not working properly so i modified little bit.
Pls see the changes made